### ISSUE TITLE
Regression with chain

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#5692]
**Describe the bug**
There seems to be a regression in 1.1.321, see example below. Code passes without errors in 1.1.320.

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
# pyright: strict


from itertools import chain
from typing import Generic, TypeVar
from collections.abc import Iterator

T = TypeVar("T")


class Container(Generic[T]):
    def __iter__(self) -> Iterator[T]:
        ...


def func(x: Container[str], y: Container[int]):
    #   error: Argument of type "Container[int]" cannot be assigned to parameter "iterables" of type "Iterable[_T@chain]" in function "__init__"
    # "Container[int]" is incompatible with protocol "Iterable[str]"
    #   "__iter__" is an incompatible type
    #     Type "() -> Iterator[int]" cannot be assigned to type "() -> Iterator[_T_co@Iterable]"
    #       Function return type "Iterator[int]" is incompatible with type "Iterator[_T_co@Iterable]"
    #         "Iterator[int]" is incompatible with "Iterator[_T_co@Iterable]" (reportGeneralTypeIssues)
    return chain(x, y)

```

**Comment by erictraut** [comment]
Thanks for the bug report. Although it's not immediately obvious, this is the same root cause as https://github.com/microsoft/pyright/issues/5673, which has already been fixed. It will be addressed in the next release. Apologies for the regression. I now have a unit test in place that checks this condition.

**Comment by erictraut** [comment]
This regression has affected several pyright users in various ways, so I decided to do a hot-fix release. I just published 1.1.322, which addresses the issue.
